packages/atomic/src/components/ipx/atomic-ipx-refine-modal/atomic-ipx-refine-modal.tsx

```diff
@@ -78,32 +78,40 @@ export class AtomicIPXRefineModal implements InitializableComponent {
     }
   }
 
+  private getAtomicModalDimensions(): DOMRect | undefined {
+    const parentIPX =
+      this.bindings.interfaceElement.querySelector('atomic-ipx-modal') ??
+      this.bindings.interfaceElement.querySelector('atomic-ipx-embedded');
+    return parentIPX!.shadowRoot
+      ?.querySelector('atomic-ipx-body')!
+      .shadowRoot?.querySelector('article[part="container"]')!
+      .getBoundingClientRect();
+  }
+
   private onAnimationFrame() {
     if (!this.isOpen) {
       return;
     }
-    if (this.dimensionChanged()) {
-      this.updateDimensions();
+    const atomicModalDimensions = this.getAtomicModalDimensions();
+    if (
+      !!atomicModalDimensions &&
+      this.dimensionChanged(atomicModalDimensions)
+    ) {
+      this.updateDimensions(atomicModalDimensions);
     }
     window.requestAnimationFrame(() => this.onAnimationFrame());
   }
 
-  private dimensionChanged() {
+  private dimensionChanged(atomicModalDimensions: DOMRect) {
```

Comment:
I didn't remember that we had all that dimension logic in the refine modal... copying it over to IPX may be the reason you had issues with the animation and `overflow: hidden` not working. I wonder what would happen if we removed all of the dimension code from IPX (including the `top`, `left`, `width` and `height` properties below).

Reply:
@btaillon not sure what you mean, where do you want me to try to remove those values from ?

Reply:
Basically, there's a bunch methods related to dimensions on the `atomic-ipx-refine-modal` and `atomic-insight-refine-modal` which aren't needed for `atomic-refine-modal`. I believe those were needed to ensure that the refine modal doesn't spill out of the boundaries in the refine page, I completely forgot about it until now. With the new option to make `atomic-modal` not expand beyond its own size, these are likely no longer necessary, and may even have caused the issue you saw before.

> where do you want me to try to remove those values from ?

https://github.com/coveo/ui-kit/blob/51b378972e506ecd67b791f2b3e8dd6ca6f974eb/packages/atomic/src/components/ipx/atomic-ipx-refine-modal/atomic-ipx-refine-modal.tsx#L159-L160

Reply:
huh, looks like removing it from `atomic-ipx-refine-modal` maintains all functionality !